core/src/main/java/org/apache/iceberg/ManifestFilterManager.java

```diff
@@ -185,6 +198,16 @@ List<ManifestFile> filterManifests(Schema tableSchema, List<ManifestFile> manife
       return ImmutableList.of();
     }
 
+    // Use the current set of referenced manifests as a source of truth when it's a subset of all
+    // manifests and all removals which were performed reference manifests.
+    // If a manifest is not in the trusted referenced set, filtering can be skipped
+    // assuming that the manifest does not have any live entries or aged out deletes
+    Set<String> manifestLocations =
+        manifests.stream().map(ManifestFile::path).collect(Collectors.toSet());
+    boolean trustReferencedManifests =
+        allDeletesReferenceManifests
+            && manifestLocations.containsAll(manifestsReferencedForDeletes);
```

Comment:
Instead of making `trustReferencedManifests` an instance variable, I made it an argument that gets plumbed through. I tried upleveling more state to be passed through but the challenge is it ends up being more convoluted since there are cases (e.g. aged out deletes or new manifests with live entries) which still need filtering even though the trusted set does not reference those manifests. Upleveling more requires upleveling even more of the filter logic for these other cases.

Reply:
Let me know if you think the new approach is easier to read or you prefer going back to the more instance state based approach @rdblue . I think keeping it all in a single boolean which includes the fact that all the deletes referenced manifests makes it a little bit better.